File: solutions/uri/1068/1068.cpp
#include <iostream>
#include <string>

int
    main() {
    int c;

    std::string s;

    while (std::cin >> s) {
        c = 0;

        for (unsigned int i = 0; i < s.length(); ++i) {
            if (s.at(i) == ')') {
                c--;
            } else if (s.at(i) == '(') {
                c++;
            }

            if (c < 0)
                break;
        }

        if (c == 0) {
            std::cout << "correct" << std::endl;
        } else {
            std::cout << "incorrect" << std::endl;
        }
    }

    return 0;
}

File: solutions/beecrowd/1068/1068.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

while line := input().strip():
    c = 0

    for character in line:
        if character == ')':
            c -= 1
        if character == '(':
            c += 1

        if c < 0:
            break

    output(f'{"correct" if c == 0 else "incorrect"}\n')
